sway-parse/src/item/item_struct.rs

```diff
@@ -7,6 +7,7 @@ use sway_types::Spanned;
 
 impl Parse for ItemStruct {
     fn parse(parser: &mut Parser) -> ParseResult<ItemStruct> {
+        let visibility = parser.take();
```

Comment:
Here was my solution based on how `item_enum` looked, and that it was passing the test that `struct` was not. Let me know if this needs to change, I'm not well versed in parser-land requirements. @mohammadfawaz

Reply:
This does seem reasonable. I'm just trying to understand how this wasn't causing issues. I quickly tried importing a private struct (without this change) and I couldn't so clearly the `pub` keyword is being respected somewhere.

Reply:
Not sure how you could recreate this scenario other than adding the `pub` keyword like I did to the test I've changed here

Reply:
I think this would only be important in instances where you are trying to parse the `struct` standalone or as a snippet, like I am in the tests and in `forc doc`

Reply:
Hmm notice that the visibility qualifier is collected in https://github.com/FuelLabs/sway/blob/master/sway-parse/src/item/mod.rs#L37-L39 which is a bit weird 🤔

Reply:
Should I open an issue and put this PR on hold?

Reply:
I would say just continue with this PR as is. We'll clean up all that stuff later.

Reply:
OK, it's working as intended if you wouldn't mind a review!